google-api-client/pom.xml

```diff
@@ -110,7 +111,6 @@
     <dependency>
       <groupId>com.google.http-client</groupId>
       <artifactId>google-http-client-gson</artifactId>
-      <scope>test</scope>
```

Comment:
Yes, you should take the JsonFactory as a parameter to CloudShellCredential, and use the one that getApplicationDefault has had passed in. If you want to have a default value on CloudShellCredential for cases where people use it directly, use Util.getDefaultJsonFactory().